crawler/crawler-ftp/src/test/java/fr/pilato/elasticsearch/crawler/fs/crawler/ftp/FileAbstractorFTPTest.java

```diff
@@ -0,0 +1,163 @@
+    private final String pass = "password";
+
+    @Before
+    public void setup() {
+        // it doesn't support utf-8
+        fakeFtpServer = new FakeFtpServer();
+        fakeFtpServer.setServerControlPort(5968);
+        fakeFtpServer.addUserAccount(new UserAccount(user, pass, home));
+        FileSystem fileSystem = new UnixFakeFileSystem();
+
+        fileSystem.add(new DirectoryEntry(home));
+        fileSystem.add(new FileEntry(home + "/foo.txt", "文件名不支持中文"));
+        fileSystem.add(new FileEntry(home + "/bar.txt", "bar"));
+
+        fileSystem.add(new DirectoryEntry(home + "/buzz"));
+        fileSystem.add(new FileEntry(home + "/buzz/hello.txt", "hello"));
+        fileSystem.add(new FileEntry(home + "/buzz/world.txt", "world"));
+
+        fakeFtpServer.setFileSystem(fileSystem);
+        fakeFtpServer.start();
+    }
+
+    @After
+    public void teardown() {
+        fakeFtpServer.stop();
+    }
+
+    @Test
+    public void testConnectToFakeFTPServer() throws Exception {
+        int port = fakeFtpServer.getServerControlPort();
+        FsSettings fsSettings = FsSettings.builder("fake")
+                .setServer(
+                        Server.builder()
+                                .setHostname("localhost")
+                                .setUsername(user)
+                                .setPassword(pass)
+                                .setPort(port)
+                                .build()
+                )
+                .build();
+
+        FileAbstractorFTP ftp = new FileAbstractorFTP(fsSettings);
+        ftp.open();
+        boolean exists = ftp.exists(home);
+        assertThat(exists, is(true));
+        Collection<FileAbstractModel> files = ftp.getFiles(home);
+        assertThat(files.size(), is(3));
+
+        for (FileAbstractModel file : files) {
+            if (file.isDirectory()) {
+                assertThat(file.getName(), is("buzz"));
+                Collection<FileAbstractModel> subDirFiles = ftp.getFiles(file.getFullpath());
+                assertThat(subDirFiles.size(), is(2));
+                logger.debug("Found {} files in sub dir", subDirFiles.size());
+                for (FileAbstractModel subDirFile : subDirFiles) {
+                    try (InputStream inputStream = ftp.getInputStream(subDirFile)) {
+                        String content = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+                        logger.debug("[{}] - {}: {}", file.getName(), subDirFile.getName(), content);
+                    }
+                }
+            } else {
+                try (InputStream inputStream = ftp.getInputStream(file)) {
+                    String content = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+                    logger.debug(" - {}: {}", file.getName(), content);
+                }
+            }
+        }
+
+        ftp.close();
+    }
+
+    @Test @Ignore
+    public void testConnectToFTPServer() throws Exception {
```

Comment:
Why this one is ignored?

Reply:
It seems like `MockFtpServer` doesn't support `utf-8` so I'm using this to do a real crawling about Chinese characters. Will add some comments on it.